arroyo/processing/processor.py

```diff
@@ -169,6 +170,10 @@ def __init__(
             DlqPolicyWrapper(dlq_policy) if dlq_policy is not None else None
         )
 
+        self.__last_run_log_ts = time.time()  # This is for throttling the logging of each run loop per-consumer
+        self.__last_pause_ts = None
```

Comment:
you can simplify your logic by setting all of these to `0`, then you don't have to check for `None`